spring-integration-core/src/main/java/org/springframework/integration/support/json/Jackson2JsonObjectMapper.java

```diff
@@ -137,7 +157,8 @@ protected JavaType extractJavaType(Map<String, Object> javaTypes) throws Excepti
 		JavaType contentClassType = this.createJavaType(javaTypes, JsonHeaders.CONTENT_TYPE_ID);
 		if (classType.getKeyType() == null) {
 			return this.objectMapper.getTypeFactory()
-					.constructCollectionType((Class<? extends Collection<?>>) classType.getRawClass(), contentClassType);
+					.constructCollectionType((Class<? extends Collection<?>>) classType
+							.getRawClass(), contentClassType);
```

Comment:
More IDEA weirdness - did you upgrade again?

Reply:
Well, I get this option since I moved to new Mac Book.
So, it is independent of the upgrade.
I'll switch off an option to *wrap if longer then 120*.

Reply:
**It's something else...

```
			return this.objectMapper.getTypeFactory()
					.constructCollectionType((Class<? extends Collection<?>>) classType.getRawClass(), 
							contentClassType);
```

is wrapped at 102 so I don't know why it split at 87.**

Reply:
Well, original line for me in IDEA looks like this:

![image](https://user-images.githubusercontent.com/1071590/46696339-f007f900-cbdf-11e8-8635-5339c89ff620.png)

So, it is longer than 120, therefore during reformatting IDEA decides to wrap it on the last `.` assuming it is a method call. That's what I see in other cases you pointed, too.

I don't think it's a big deal to have a discussion now.  And unfortunately I don't see how to turn it off now 😢